RecoParticleFlow/PFSimProducer/plugins/HGCalTrackCollectionProducer.cc

```diff
@@ -0,0 +1,190 @@
+  edm::ESHandle<TrackerGeometry> tkGeom_;
+  std::array<std::string,1> hgc_names_; // 3 --> 1; extrapolate to hgcee only
+  std::array<edm::ESHandle<HGCalGeometry>,1> hgcGeometries_; // 3 --> 1; extrapolate to hgcee only
+  std::array<std::vector<ReferenceCountingPointer<BoundDisk> >,1> plusSurface_,minusSurface_; // 3 --> 1; extrapolate to hgcee only
+  std::unique_ptr<PropagatorWithMaterial> mat_prop_;
+
+  float diskOuterRadius_;
+  float diskInnerRadius_;
+};
+
+HGCalTrackCollectionProducer::HGCalTrackCollectionProducer(const edm::ParameterSet & iConfig) :
+  src_(consumes<edm::View<reco::PFRecTrack> >(iConfig.getParameter<edm::InputTag> ("src"))),
+  debug_(iConfig.getParameter<bool>("debug")),
+  DPtovPtCut_(iConfig.getParameter<std::vector<double> >("DPtOverPtCuts_byTrackAlgo")),
+  NHitCut_(iConfig.getParameter<std::vector<unsigned> >("NHitCuts_byTrackAlgo")),
+  useIterTracking_(iConfig.getParameter<bool>("useIterativeTracking"))
+  //  _useFirstLayerOnly(iConfig.getParameter<bool>("UseFirstLayerOnly")) // always true now
+{
+
+  if (debug_) std::cout << " HGCalTrackCollectionProducer::HGCalTrackCollectionProducer " << std::endl;
+
+  const edm::ParameterSet& geoconf = iConfig.getParameterSet("hgcalGeometryNames");
+  hgc_names_[0] = geoconf.getParameter<std::string>("HGC_ECAL");
+  // 3 --> 1; extrapolate to hgcee only
+  //  hgc_names_[1] = geoconf.getParameter<std::string>("HGC_HCALF"); 
+  //  hgc_names_[2] = geoconf.getParameter<std::string>("HGC_HCALB");
+
+  produces<reco::PFRecTrackCollection>("TracksInHGCal");
+  produces<reco::PFRecTrackCollection>("TracksNotInHGCal");
+
+}
+
+// From https://github.com/cms-sw/cmssw/blob/CMSSW_6_2_X_SLHC/RecoParticleFlow/PFClusterProducer/src/HGCClusterizer.cc#L441-L447 and beyond
+// TODO: we only need the front of the calorimeter, so modify this
+void HGCalTrackCollectionProducer::beginLuminosityBlock(const edm::LuminosityBlock& lumi, const edm::EventSetup& es) {
+  constexpr float m_pion = 0.1396;
+  // get dependencies for setting up propagator  
+  es.get<IdealMagneticFieldRecord>().get(bField_);
+  es.get<TrackerDigiGeometryRecord>().get(tkGeom_);
+  // get HGC geometries (assume that layers are ordered in Z!)
+  for( unsigned i = 0; i < hgcGeometries_.size(); ++i ) {
+    es.get<IdealGeometryRecord>().get(hgc_names_[i],hgcGeometries_[i]);
+  }
+  
+  // make propagator
+  mat_prop_.reset( new PropagatorWithMaterial(alongMomentum, m_pion, bField_.product()) );
+  // setup HGC layers for track propagation
+  Surface::RotationType rot; //unit rotation matrix
+  for( unsigned i = 0; i < hgcGeometries_.size(); ++i ) {
+    minusSurface_[i].clear();
+    plusSurface_[i].clear();
+    const HGCalDDDConstants &dddCons=hgcGeometries_[i]->topology().dddConstants();
+    std::map<float,float> zrhoCoord;
+    std::map<float,float> innerRadiusCoord;
+    const auto& firstLayerIt = dddCons.getTrForms().back();
+    //const auto& firstmod = dddCons.getModule(0,true,true);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant